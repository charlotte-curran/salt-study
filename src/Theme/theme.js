export const SaltTheme = {
  color: {
    dark: "#222324",
    blue: "#92b3d2",
    white: "rgba(255,255,255, 1)",
    orange: "#ff7a61",
    shadow: "#545353",
    address: "rgba(255, 255, 255, 0.4)",
    navbar: "rgba(34, 35, 36, 0.7)"
  },
  fonts: {
    futuraPt: '"futura-pt", sans-serif',
    raleway: '"raleway", sans-serif'
  }
};
