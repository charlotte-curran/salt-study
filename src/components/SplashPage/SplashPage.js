import React from "react";
import { useLocaleContext } from "../../Context/localeContext";
import ButtonWhite from "../ButtonWhite";
import plusImage from "../../images/saltplus.svg";
import backgroundImage from "../../images/x-clusion-min.png";
import { saltStudyText } from "../../TextContent/saltstudy";
import {
  PlusBottomLeft,
  PlusTopLeft,
  PlusBottomRight,
  PlusTopRight
} from "../SquareOne/SquareOne.styles";
import {
  Wrapper,
  InnerWrapper,
  Header,
  Main,
  ButtonWrapper,
  BlueText
} from "./SplashPage.styles";

const SplashPage = () => {
  const { swapLocale } = useLocaleContext();
  return (
    <Wrapper background={backgroundImage}>
      <PlusTopLeft src={plusImage} />
      <PlusBottomLeft src={plusImage} />
      <PlusTopRight src={plusImage} />
      <PlusBottomRight src={plusImage} />
      <InnerWrapper>
        <Header>
          {saltStudyText.splashPage.header[0]}
          <BlueText>{saltStudyText.splashPage.header[1]}</BlueText>
          {saltStudyText.splashPage.header[2]}
        </Header>
        <Main>{saltStudyText.splashPage.main}</Main>
        <ButtonWrapper>
          <ButtonWhite
            isFooter={true}
            onClick={() => {
              swapLocale(1);
            }}
          >
            Stockholm
          </ButtonWhite>
          <ButtonWhite
            isFooter={true}
            onClick={() => {
              swapLocale(2);
            }}
          >
            Amsterdam
          </ButtonWhite>
          <ButtonWhite
            isFooter={true}
            onClick={() => {
              swapLocale(3);
            }}
          >
            Oslo
          </ButtonWhite>
        </ButtonWrapper>
      </InnerWrapper>
    </Wrapper>
  );
};

export default SplashPage;
