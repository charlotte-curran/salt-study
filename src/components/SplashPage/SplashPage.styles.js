import styled from "styled-components";

export const Wrapper = styled.div`
  background: url(${({ background }) => background}),
    ${({ theme }) => theme.color.dark};
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const InnerWrapper = styled.div`
  width: 80vw;
  height: 70vh;
  padding: 100px 0;
  border-radius: 2px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  color: ${({ theme }) => theme.color.white};
  font-family: ${({ theme }) => theme.fonts.futuraPt};
  text-align: center;

  @media (min-width: 350px) and (min-height: 700px) {
    padding: 200px 0;
  }

  /* targeting landscape oriented phones */
  @media only screen and (min-width: 480px) and (max-width: 896px) and (min-height: 320px) and (max-height: 414px) and (orientation: landscape) {
    padding: 40px 0;
  }
`;

export const Header = styled.div`
  text-transform: uppercase;
  font-weight: 800;
  font-size: 36px;

  @media (min-width: 700px) {
    font-size: 48px;
  }
`;

export const BlueText = styled.span`
  color: ${({ theme }) => theme.color.blue};
`;

export const Main = styled.div`
  font-size: 26px;

  @media (min-width: 700px) {
    font-size: 30px;
  }
`;

export const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 200px;

  button {
    width: 100%;
  }
`;
