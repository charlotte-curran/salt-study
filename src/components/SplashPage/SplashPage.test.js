import React from "react";
import { create } from "react-test-renderer";
import SplashPage from "./SplashPage";
import { ThemeProvider } from "styled-components";
import { SaltTheme } from "../../Theme/theme";
import LocaleProvider from "../../Context/localeContext";

describe("Home", () => {
  test("matches the snapshot", () => {
    const splashPage = create(
      <LocaleProvider>
        <ThemeProvider theme={SaltTheme}>
          <SplashPage />
        </ThemeProvider>
      </LocaleProvider>
    );
    expect(splashPage.toJSON()).toMatchSnapshot();
  });
});
