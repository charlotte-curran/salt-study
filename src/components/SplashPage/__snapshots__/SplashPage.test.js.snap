// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div
  className="sc-dnqmqq llekxD"
>
  <img
    className="sc-ifAKCX jBVcFo"
    src="saltplus.svg"
  />
  <img
    className="sc-bZQynM iACVWl"
    src="saltplus.svg"
  />
  <img
    className="sc-EHOje eyCOiF"
    src="saltplus.svg"
  />
  <img
    className="sc-gzVnrw iDSDOu"
    src="saltplus.svg"
  />
  <div
    className="sc-iwsKbI bMqjPQ"
  >
    <div
      className="sc-gZMcBi eUpego"
    >
      Welcome to 
      <span
        className="sc-gqjmRU hQZdwx"
      >
        salt
      </span>
      !
    </div>
    <div
      className="sc-VigVT bzhQOw"
    >
      Pick your location:
    </div>
    <div
      className="sc-jTzLTM jZbgRH"
    >
      <button
        className="sc-bdVaJa ipMqzT"
        onClick={[Function]}
      >
        Stockholm
      </button>
      <button
        className="sc-bdVaJa ipMqzT"
        onClick={[Function]}
      >
        Amsterdam
      </button>
      <button
        className="sc-bdVaJa ipMqzT"
        onClick={[Function]}
      >
        Oslo
      </button>
    </div>
  </div>
</div>
`;
